runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingStepMetricsContainer.java

```diff
@@ -402,12 +387,17 @@ Iterable<PerStepNamespaceMetrics> extractPerWorkerMetricUpdates() {
           counters.put(k, val);
         });
 
-    perWorkerGauges.forEach(
+    gauges.forEach(
         (k, v) -> {
-          Long val = v.getCumulative().value();
-          gauges.put(k, val);
-          v.reset();
+          // Check if metric name has the per worker label set
+          if (k.getLabels().containsKey("PER_WORKER_METRIC")
```

Comment:
I think this could be optimized by keeping the perWorkerGauges and populating it by doing this labels check when the gauge is created.

Could add a comment about possible optimization and do it when fixing the other metric types.

Reply:
Oh that makes sense, so we don't have to check each time we process an update.